src/node/setupServer.ts

```diff
@@ -1,15 +1,137 @@
 import { ClientRequestInterceptor } from '@mswjs/interceptors/lib/interceptors/ClientRequest'
 import { XMLHttpRequestInterceptor } from '@mswjs/interceptors/lib/interceptors/XMLHttpRequest'
-import { createSetupServer } from './createSetupServer'
+import { SetupApi } from '../createSetupApi'
+import { RequestHandler } from '../handlers/RequestHandler'
+import { LifeCycleEventsMap, SharedOptions } from '../sharedOptions'
+import { RequiredDeep } from '../typeUtils'
+import { mergeRight } from '../utils/internal/mergeRight'
+import { bold } from 'chalk'
+import { MockedRequest } from '../utils/request/MockedRequest'
+import { handleRequest } from '../utils/handleRequest'
+import {
+  IsomorphicResponse,
+  MockedResponse as MockedInterceptedResponse,
+} from '@mswjs/interceptors'
+
+export type ServerLifecycleEventsMap = LifeCycleEventsMap<IsomorphicResponse>
+
+const DEFAULT_LISTEN_OPTIONS: RequiredDeep<SharedOptions> = {
+  onUnhandledRequest: 'warn',
+}
+
+/**
+ * Concrete class to implement the SetupApi for the node server environment, uses both the ClientRequestInterceptor
+ * and XMLHttpRequestInterceptor.
+ */
+export class SetupServerApi extends SetupApi<ServerLifecycleEventsMap> {
+  private resolvedOptions: RequiredDeep<SharedOptions>
+
+  constructor(handlers: RequestHandler[]) {
+    super(
+      [ClientRequestInterceptor, XMLHttpRequestInterceptor],
+      'server-setup',
+      handlers,
+    )
+
```

Comment:
I am contemplating if I should make the interceptors in the concrete class dynamic so that whoever uses the server-like environment setup can pass in their interceptors and reuse this class

```ts 
class SetupServerApi extends SetupApi<ServerLifecycleEventsMap> {
    constructor(handlers: RequestHandler[], interceptors: ...[]) {
       super(interceptors, 'server-setup', handlers)
    }
}

```

Then to use this..

```ts
  export const setupServer = (...handlers: RequestHandler[]) => {
    return new SetupServerApi(handlers, [ClientRequestInterceptor, XMLHttpRequestInterceptor])
  }
```

Reply:
I think the entire interceptors concept must be moved away from the base `SetupApi` and implemented as a custom thing on those child classes that need interceptors (Node and React Native ones). This would also make the usage of the base class more streamlined as there are setup APIs that do not need any interceptors at all, like `setupWorker`. 

This way the base class is basically an internal representation of handlers, internal emitters (and even handling), and the public methods that are shared across all setup APIs.